soc/arm/st_stm32/stm32h7/mpu_regions.c

```diff
@@ -12,6 +12,15 @@ static const struct arm_mpu_region mpu_regions[] = {
 					 REGION_FLASH_ATTR(REGION_FLASH_SIZE)),
 	MPU_REGION_ENTRY("SRAM", CONFIG_SRAM_BASE_ADDRESS,
 					 REGION_RAM_ATTR(REGION_SRAM_SIZE)),
+	/*
+	 * System memory declared as 'Execute-Never' inhibits the speculative fetch,
+	 * preventing the RDSERR
+	 */
+	MPU_REGION_ENTRY("SYSTEM", 0x1FF00000,
+					{ (STRONGLY_ORDERED_SHAREABLE |
+					REGION_512K |
+					FULL_ACCESS_Msk) }),
```

Comment:
Wouldn't `MPU_RASR_XN_Msk | P_RW_U_NA_Msk`  instead of `FULL_ACCESS_Msk` be more appropriate? Then it would be the same as `REGION_RAM_ATTR` above. But generally `STRONGLY_ORDERED_SHAREABLE` solves all issues in our application.

Reply:
yes, this also removes the RDSERR (even if do not really catch the difference)

Reply:
My understanding is that with `MPU_RASR_XN_Msk` we avoid to execute code from this region and with `P_RW_U_NA_Msk` we avoid that a user mode thread can access it. Both should not happen in my point of view and the configuration should therefore be a bit more secure.